packages/strapi-helper-plugin/lib/internals/generators/route/index.js

```diff
@@ -55,27 +48,58 @@ module.exports = {
     },
   }],
 
-  // Add the route to the routes.js file above the error route
-  // TODO smarter route adding
   actions: (data) => {
-    const actions = [];
-    if (reducerExists(data.component)) {
-      data.useSagas = sagasExists(data.component); // eslint-disable-line no-param-reassign
-      actions.push({
-        type: 'modify',
-        path: '../../../../../admin/src/routes.js',
-        pattern: /(\s{\n\s{0,}path: '\*',)/g,
-        template: trimTemplateFile('routeWithReducer.hbs'),
-      });
-    } else {
-      actions.push({
-        type: 'modify',
-        path: '../../../../../admin/src/routes.js',
-        pattern: /(\s{\n\s{0,}path: '\*',)/g,
-        template: trimTemplateFile('route.hbs'),
-      });
-    }
+    const replaceFile = () => {
```

Comment:
The construction of this method is very difficult to read. Many try/catch with unscoped variables. Also, you could use `fs.access` to test if a file exists instead of `fs.statsSync`? I'm not sure you need to check if this is a file or not. I can have wrong on this one, I'm just asking...